test/unit-test/test_lib/test_analysis/test_user.py

```diff
@@ -1,34 +1,146 @@
 import unittest
 from lib.analysis import user
+import lib.config as config
+from lib.in_out import reader, saver
+import lib.nickTracker as nickTracker
+from mock import patch
+import networkx as nx
 import lib.util as util
 import os
+import StringIO
+import sys
+import filecmp
 
 current_directory = os.path.dirname(os.path.realpath(__file__))
 
 class UserTest(unittest.TestCase):
 
     def setUp(self):
-        self.log_data = util.load_from_disk(current_directory+ "/data/log_data")
-        self.nicks_hash = util.load_from_disk(current_directory+ "/data/nicks_hash")
+        #./data/user_test/log_data has data data for the following configuraton:
+        #   STARTING_DATE = "2013-01-01"
+        #   ENDING_DATE = "2013-01-01"
+        #   CHANNEL_NAME = ["#kubuntu"]
+        self.log_data = util.load_from_disk(current_directory+"/data/user_test/log_data")
+        self.nicks = util.load_from_disk(current_directory+ "/data/user_test/nicks")
+        self.nick_same_list = util.load_from_disk(current_directory+"/data/user_test/nick_same_list")
+        self.build_graphs_list = util.load_from_disk(current_directory + "/data/user_test/build_graphs_list")
+        self.rec_list_splice_list = util.load_from_disk(current_directory + "/data/user_test/rec_list_splice_list")
+        self.rec_list_splice_util_call_counter = 0
+        self.build_graphs_util_call_counter = 0
+
 
     def tearDown(self):
         self.log_data = None
-        self.nicks_hash = None
+        self.nicks = None
+        self.nick_same_list = None
+        self.build_graphs_list = None
+        self.rec_list_splice_list = None
+        self.rec_list_splice_util_call_counter = None
+        self.build_graphs_util_call_counter = None
+
+
+    def __build_graphs_util(nick_sender, nick_receiver, time, year, month, day, day_graph, aggr_graph):
```

Comment:
Please change the signature of the function to return an object and test based on that object.